ERCS/erc-7930.md

```diff
@@ -0,0 +1,248 @@
+
+The current standard ethereum address representation defined in [ERC-55], with _optional_ retrofitted checksums implemented via selective capitalization of `a-f` characters, poses the following challenges:
+
+- Verification of the checksum by a client is optional, meaning that mistyped or erroneous addresses may be accepted by these clients. There is no way for a user to distinguish a correct address from an incorrect one processed by a client that does not do validation.
+- Addresses do not include any indication of the chain to which the address applies. This means chain information must be expressed out of band, and introduces the risk of an address being mistakenly used on a chain it is not valid for. This risk is particularly pronounced for addresses that represent smart contracts.
+- The existing address format provides no mechanism for extension, meaning that there is no natural method to add these features to the existing address format.
+
+Interoperable Addresses build on insights from [ERC-7828], [CAIP-10] and [CAIP-50], offering a unified format which combines:
+- Binding chain specificity (via explicit chain identifiers) to the raw address.
+- Compact & canonical binary format for use on cross-chain message passing and intent declaration.
+- Checksums for name collision mitigation & user error prevention.
+
+Furthermore, this proposal adds a few features to make addresses interoperable with infrastructure at the edges of and beyond the ethereum ecosystem, by supporting the representation of addresses of non-evm chains as well, for the benefit of cross-chain liquidity movements and cross-chain message passing.
+
+### Guarantees this standard aims to provide
+- Any future Interoperable Address will be trivially convertible to Interoperable Address v1 (specified below), which, together with the point below, makes them a canonical representation for users who need to treat them opaquely.
+- Two Interoperable Addresses, converted to this canonical version, will be bitwise-equal if and only if they represent the same _target address_.
+- Checksums are short enough to be compared by humans, and allow for easy differentiation of _target addresses_ independently of any extra data the Interoperable Address may contain.
+
+## Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Concepts
+Interoperable Address
+: A binary payload which unambiguously identifies a target address and allows conversion to a human-readable name.
+
+Interoperable Name
+: A string representation of an interoperable address, meant to be used by humans.
+
+Target address
+: The (address, chain ID) pair a particular Interoperable Address points to.
```

Comment:
The definitions have a circular reference issue where "Interoperable Address" references "target address" before it's defined. Consider this revision to maintain the same order while fixing the circularity:

```suggestion
Interoperable Address
: Binary encoding of a (chain, address) pair with version and checksum.

Interoperable Name
: Human-readable text representation of an interoperable address.

Target Address
: A (chain, address) pair identifying a specific account on a specific blockchain.
```

This way each definition builds on concepts already established rather than forward-referencing.